src/Tribe/Assets.php

```diff
@@ -159,44 +159,50 @@ public function register() {
 			)
 		);
 
-		// FrontEnd
-		$api_url = 'https://maps.google.com/maps/api/js';
-		$api_key = tribe_get_option( 'google_maps_js_api_key' );
-
-		if ( ! empty( $api_key ) && is_string( $api_key ) ) {
-			$api_url = sprintf( 'https://maps.googleapis.com/maps/api/js?key=%s', trim( $api_key ) );
+		// Some Google Maps API-specific scripts that should only load when a non-default API key is present.
+		if ( ! tribe_is_using_basic_gmaps_api() ) {
+
+			// FrontEnd
+			$api_url = 'https://maps.google.com/maps/api/js';
+			$api_key = tribe_get_option( 'google_maps_js_api_key', Tribe__Events__Google__Maps_API_Key::$default_api_key );
+
+			if ( ! empty( $api_key ) && is_string( $api_key ) ) {
+				$api_url = sprintf( 'https://maps.googleapis.com/maps/api/js?key=%s', trim( $api_key ) );
+			}
+
+			/**
+			 * Allows for filtering the embedded Google Maps API URL.
+			 *
```

Comment:
Add `@since`

Reply:
I won't add `@since TBD` here, since this filter is preexisting and is not new with these changes. I will, however, add `@since ??` per our processes.

Reply:
Ok